lang/allocator/README.md

```diff
@@ -0,0 +1,21 @@
+# MobileCoin SGX: Allocator
+
+[![Project Chat][chat-image]][chat-link]<!--
+-->![License][license-image]<!--
+-->![Target][target-image]<!--
+-->[![Crates.io][crate-image]][crate-link]<!--
+-->[![Docs Status][docs-image]][docs-link]<!--
+-->[![Dependency Status][deps-image]][deps-link]
+
+Provide allocator for use in an SGX enclave
```

Comment:
Note that this is the lang item, the allocator is provided by the `mc-sgx-alloc` crate.

Reply:
updated statement similar to the cargo toml description, also reworded header